File: templates/matlab/effect_size_panel.m
function fig = effect_size_panel()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(0);
    mu1 = 0.0; mu2 = 1.2; sd = 1.0;
    d = (mu2 - mu1) / sd;
    x = linspace(min(mu1, mu2) - 4*sd, max(mu1, mu2) + 4*sd, 500);
    pdf1 = exp(-0.5*((x - mu1)/sd).^2) / (sd*sqrt(2*pi));
    pdf2 = exp(-0.5*((x - mu2)/sd).^2) / (sd*sqrt(2*pi));
    ov = min(pdf1, pdf2);
    fig = figure; hold on;
    hov = fill([x fliplr(x)], [ov zeros(size(ov))], [0.5 0.5 0.5], ...
               'FaceAlpha', 0.35, 'EdgeColor', 'none');
    h1 = plot(x, pdf1, 'Color', palette('cat',1), 'LineWidth', 1.5);
    h2 = plot(x, pdf2, 'Color', palette('cat',2), 'LineWidth', 1.5);
    ymax = max(pdf1);
    % double-headed arrow between the two means + Cohen's d label
    plot([mu1 mu2], [1 1]*ymax*1.06, 'k-', 'LineWidth', 1.1);
    plot(mu1, ymax*1.06, 'k<', 'MarkerFaceColor', 'k', 'MarkerSize', 4);
    plot(mu2, ymax*1.06, 'k>', 'MarkerFaceColor', 'k', 'MarkerSize', 4);
    text((mu1 + mu2)/2, ymax*1.10, sprintf('d = %.2f', d), ...
         'HorizontalAlignment', 'center', 'FontSize', 9);
    % benchmark ruler: small / medium / large
    bench = [0.2 0.5 0.8]; lbls = {'small', 'medium', 'large'};
    for i = 1:3
        xb = mu1 + bench(i)*sd;
        plot([xb xb], [ymax*1.00 ymax*1.03], 'Color', [0.5 0.5 0.5], 'LineWidth', 0.9);
        text(xb, ymax*0.965, lbls{i}, 'HorizontalAlignment', 'center', ...
             'FontSize', 7, 'Color', [0.5 0.5 0.5]);
    end
    plot([mu1 mu1], [0 ymax], ':', 'Color', palette('cat',1), 'LineWidth', 0.9);
    plot([mu2 mu2], [0 ymax], ':', 'Color', palette('cat',2), 'LineWidth', 0.9);
    ylim([0 ymax*1.22]);
    xlabel('outcome value'); ylabel('density'); title('Effect size: Cohen''s d');
    legend([h1 h2 hov], {'group 1', 'group 2', 'overlap'}, 'Location', 'northwest');
    grid on;
end

File: templates/python/effect_size_panel.py
"""effect_size_panel: 效应量面板（两分布重叠可视化 + Cohen's d 标尺）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(mu1=0.0, mu2=1.2, sd=1.0, title="Effect size: Cohen's d"):
    apply_theme()
    d = (mu2 - mu1) / sd
    x = np.linspace(min(mu1, mu2) - 4*sd, max(mu1, mu2) + 4*sd, 500)
    pdf1 = np.exp(-0.5*((x - mu1)/sd)**2) / (sd*np.sqrt(2*np.pi))
    pdf2 = np.exp(-0.5*((x - mu2)/sd)**2) / (sd*np.sqrt(2*np.pi))
    overlap = np.minimum(pdf1, pdf2)
    fig, ax = plt.subplots()
    ax.plot(x, pdf1, color=cycle(0), label='group 1')
    ax.plot(x, pdf2, color=cycle(1), label='group 2')
    ax.fill_between(x, 0, overlap, color='gray', alpha=0.35, label='overlap')
    ymax = pdf1.max()
    ax.annotate('', xy=(mu2, ymax*1.06), xytext=(mu1, ymax*1.06),
                arrowprops=dict(arrowstyle='<->', color='k', lw=1.1))
    ax.text((mu1 + mu2)/2, ymax*1.10, f"d = {d:.2f}", ha='center', fontsize=9)
    for bench, lbl in [(0.2, 'small'), (0.5, 'medium'), (0.8, 'large')]:
        xb = mu1 + bench*sd
        ax.plot([xb, xb], [ymax*1.00, ymax*1.03], color='gray', linewidth=0.9)
        ax.text(xb, ymax*0.965, lbl, ha='center', fontsize=7, color='gray')
    ax.axvline(mu1, color=cycle(0), linestyle=':', linewidth=0.9)
    ax.axvline(mu2, color=cycle(1), linestyle=':', linewidth=0.9)
    ax.set_ylim(0, ymax*1.22)
    ax.set_xlabel('outcome value'); ax.set_ylabel('density')
    ax.set_title(title)
    ax.legend(loc='upper left')
    ax.grid(True, linestyle=':', alpha=0.5)
    fig.tight_layout()
    return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)
